src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -3732,31 +3746,69 @@ public byte byteValueExact() {
      * @jls 5.1.3 Narrowing Primitive Conversion
      */
     @Override
-    public float floatValue(){
-        if(intCompact != INFLATED) {
+    public float floatValue() {
```

Comment:
If the total size of code were a concern, I wouldn't be adverse to floatValue() being implemented as

return (float)doubleValue();

Given the 2p+2 property between the float and double formats, BigDecimal -> double -> float and BigDecimal -> float round the same way.

Reply:
No, that would not be correct. It would be subject to [double rounding](https://en.wikipedia.org/wiki/Rounding#Double_rounding), against the spec.

For example, `BigDecimal` 1.000000059604644775390626 should round to `float` `1.0000001f`.
When going to the closest `double` and then to the closest `float`, however, it first rounds to `double` `1.0000000596046448` and the to `float` `1.0f`.